flake.nix

```diff
@@ -0,0 +1,69 @@
+{
+  description = "Example Counter – Nix flake that provides the Compact compiler per system";
+
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs/nixos-24.05";
+    nixpkgs-unstable.url = "github:NixOS/nixpkgs/nixpkgs-unstable";
+  };
+
+  outputs = inputs:
+    let
+      supportedSystems = [
+        "x86_64-linux"
+        "x86_64-darwin"
+        "aarch64-darwin"
+      ];
+      forEachSupportedSystem = f:
+        inputs.nixpkgs.lib.genAttrs supportedSystems (
+          system:
+            f {
+              pkgs = import inputs.nixpkgs { inherit system; };
+              pkgsUnstable = import inputs."nixpkgs-unstable" { inherit system; };
+              inherit system;
+            }
+        );
+    in
+    {
+      devShells = forEachSupportedSystem ({ pkgs, pkgsUnstable, system }:
+        let
+          # Compact compiler release to use
+          compactVersion = "0.24.0";
+
+          # Artifact URLs by host system
+          urlForSystem = {
+            "x86_64-linux" = "https://d3fazakqrumx6p.cloudfront.net/artifacts/compiler/compactc_${compactVersion}/compactc_v${compactVersion}_x86_64-unknown-linux-musl.zip";
+            "x86_64-darwin" = "https://d3fazakqrumx6p.cloudfront.net/artifacts/compiler/compactc_${compactVersion}/compactc_v${compactVersion}_x86_64-apple-darwin.zip";
+            "aarch64-darwin" = "https://d3fazakqrumx6p.cloudfront.net/artifacts/compiler/compactc_${compactVersion}/compactc_v${compactVersion}_aarch64-darwin.zip";
+          };
```

Comment:
https://docs.midnight.network/blog/compact-developer-tools

Can you simplify this to use the newly released compact dev tools? For example here's the PR where we started using them in the CI: https://github.com/midnightntwrk/example-counter/pull/1153